### PR TITLE
feat: add API logging to client

### PR DESCRIPTION
**Pull request by tysonholub**
This PR adds logging to the client API calls and additionally the API request method to the client Response

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/python-http-client?pullRequest=123) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/python-http-client?pullRequest=123) before we can accept your contribution.<br/><hr/>**Tyson Holub** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/python-http-client?pullRequest=123) it.</sub>

**Comment by tysonholub** [comment]
Will this PR be merged?

**Comment by thinkingserious** [comment]
Hello @tysonholub,

I'm not sure just yet, it's still in our backlog for a code review. My apologies for the delay and thank you for your patience.

With Best Regards,

Elmer

File: python_http_client/client.py [hunk 1]
```diff
@@ -171,10 +181,15 @@ def _make_request(self, opener, request, timeout=None):
         """
         timeout = timeout or self.timeout
         try:
-            return opener.open(request, timeout=timeout)
+            return (opener.open(request, timeout=timeout),
+                    request.get_method())
         except HTTPError as err:
             exc = handle_error(err)
             exc.__cause__ = None
+            _logger.info('{method} Response: {status} {body}'.format(
```
Comment (childish-sambino) [inline python_http_client/client.py:189]: Should make this a `debug` log and move to above `handle_error`, in case that fails.

**Comment by childish-sambino** [comment]
Fixes https://github.com/sendgrid/python-http-client/issues/141

**Pull request merged**